azure/packages/azure-client/src/test/AzureClient.spec.ts

```diff
@@ -45,239 +45,614 @@ describe("AzureClient", () => {
-     * each other after value is changed.
-     */
-    it("can change initialObjects value", async () => {
-        const { container } = await client.createContainer(schema);
-        const containerId = await container.attach();
-        await new Promise<void>((resolve) => {
-            container.once("connected", () => {
-                resolve();
+            const mdo3 = initialObjectsCreate.mdo3 as TestDataObject2;
+            mdo3.increment();
+            mdo3.increment();
+            mdo3.increment();
+
+            assert.strictEqual(mdo2.value, 1);
+            assert.strictEqual(mdo3.value, 3);
+
+            const containerId = await container.attach();
+            await new Promise<void>((resolve) => {
+                container.once("connected", () => {
+                    resolve();
+                });
             });
+
+            const { container: containerGet } = await client.getContainer(containerId, doSchema);
+            const initialObjectsGet = containerGet.initialObjects;
+            const mdo2get = initialObjectsGet.mdo2 as TestDataObject2;
+            const mdo3get = initialObjectsGet.mdo3 as TestDataObject2;
+
+            assert.strictEqual(mdo2get.value, 1);
+            assert.strictEqual(mdo3get.value, 3);
+
+            mdo3get.increment();
+            mdo3get.increment();
+            assert.strictEqual(mdo3get.value, 5);
         });
 
-        const initialObjectsCreate = container.initialObjects;
-        const map1Create = initialObjectsCreate.map1 as SharedMap;
-        map1Create.set("new-key", "new-value");
-        const valueCreate: string | undefined = await map1Create.get("new-key");
+        /**
+         * Scenario: test if the optional schema parameter, dynamicObjectTypes (custom data objects),
+         * can be added during runtime and be returned by the container.
+         *
+         * Expected behavior: added loadable object can be retrieved from the container. Loadable
+         * object's id and container config ID should be identical since it's now attached to
+         * the container.
+         */
+        it("can create/add loadable objects (custom data object) dynamically during runtime", async () => {
+            const dynamicSchema: ContainerSchema = {
+                initialObjects: {
+                    map1: SharedMap,
+                },
+                dynamicObjectTypes: [TestDataObject],
+            };
+
+            const { container } = await client.createContainer(dynamicSchema);
 
-        const { container: containerGet } = await client.getContainer(containerId, schema);
-        const map1Get = containerGet.initialObjects.map1 as SharedMap;
-        const valueGet: string | undefined = await mapWait(map1Get, "new-key");
-        assert.strictEqual(valueGet, valueCreate, "container can't change initial objects");
+            const newPair = await container.create(TestDataObject);
```

Comment:
What does newPair mean here?  Maybe newDO instead?

Reply:
It had a pair in this test case before it was kicked out :) changed the name.